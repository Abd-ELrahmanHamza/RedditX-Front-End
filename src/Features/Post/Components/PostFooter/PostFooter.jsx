import { Container, Comment, MyDropdown } from "./PostFooter.styled";

import { TfiComment } from "react-icons/tfi";
import { VscGift } from "react-icons/vsc";
import { IoMdShareAlt } from "react-icons/io";
import { TbBadge } from "react-icons/tb";
import { HiOutlineDotsHorizontal } from "react-icons/hi";
import { BiHide } from "react-icons/bi";
import { FiFlag } from "react-icons/fi";

import { hidePost } from "Features/Post/Services/postActions";
import useFetchFunction from "Hooks/useFetchFunction";

import { useAuth } from "Features/Authentication/Contexts/Authentication";
import submitSpam from "Features/Post/Services/submitSpam";

/**
 * PostFooter Component that is in the side of Post
 * @returns {React.Component} PostFooter component
 */
const PostFooter = ({ post, setMakeHidden }) => {
  const auth = useAuth();

  /**
   * Custom hook
   */
  const [data, error, isLoading, dataFetch] = useFetchFunction();

  const handleHide = () => {
    setMakeHidden(true);
    hidePost(
      dataFetch,
      {
        linkID: `t3_${post._id}`,
      },
      auth.getToken()
    );
  };
  const handleSpam = () => {
    submitSpam(
      dataFetch,
      {
        linkID: `t3_${post._id}`,
        spamText: "I found that this content is showing violence",
        spamType: "violent content",
      },
      auth
    );
  };
  return (
    <Container>
      <Comment onClick={(event) => {}}>
        <TfiComment /> {post.commentsNumber} Comments
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <VscGift /> Award
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <IoMdShareAlt /> Share
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <TbBadge /> Save
      </Comment>
      <Comment
        onClick={(event) => {
          event.stopPropagation();
        }}
      >
        <MyDropdown className="d-inline mx-2" autoClose="outside">
          <MyDropdown.Toggle id="dropdown-autoclose-outside">
            <HiOutlineDotsHorizontal />
          </MyDropdown.Toggle>

          <MyDropdown.Menu>
            <MyDropdown.Item href="#">
              <span onClick={handleHide}>
                <BiHide /> Hide
              </span>
            </MyDropdown.Item>
            <MyDropdown.Item href="#">
              <span onClick={handleSpam}>
                <FiFlag /> Report
              </span>
            </MyDropdown.Item>
          </MyDropdown.Menu>
        </MyDropdown>
      </Comment>
    </Container>
  );
};

export default PostFooter;
