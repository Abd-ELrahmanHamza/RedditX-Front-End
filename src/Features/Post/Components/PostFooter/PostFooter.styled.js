import styled from "styled-components";
import Dropdown from "react-bootstrap/Dropdown";

export const Container = styled.div`
  display: flex;
  justify-content: start;
  align-items: center;
  background-color: white;
  padding: 0 20px 20px;
  gap: 2%;
  max-width: 100%;
`;
export const Comment = styled.span`
  color: gray;
  padding: 5px 10px;
  cursor: pointer;
  svg {
    font-size: 24px;
  }

  &:hover {
    background-color: #cccccc2b;
  }
`;
export const MyDropdown = styled(Dropdown)`
 
  .dropdown-menu.show {
    padding: 0;
    overflow: hidden;
    box-shadow: 0px 0px 10px -2px grey;
    
  }

  .dropdown-item {
    
    padding-right: 0!important;
    color: gray;
    display: flex;
    align-items: center;

    span{
        margin: 0;
    }

    &:hover {
      background-color: #5cc0f147;
      color: black !important;
    }
  }

  button {
    background-color: transparent !important;
    border-color: transparent !important;
    color: gray;

    &:after {
      display: none;
    }
  }

  &:hover {
    button {
      background-color: transparent;
      border-color: transparent;
      color: gray;
    }
  }
`;
