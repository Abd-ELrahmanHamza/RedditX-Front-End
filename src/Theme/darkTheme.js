/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
