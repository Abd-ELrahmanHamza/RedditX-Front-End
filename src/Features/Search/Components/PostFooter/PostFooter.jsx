import { Footer } from "./PostFooter.styled";

/**
 * Component that contains the Post Header Footer.
 *
 * @Component
 * @param {object} postfooter - the post footer object
 * @returns {React.Component}
 */
var abbreviate = require("number-abbreviate");
const PostFooter = ({ postfooter }) => {
  if (postfooter) {
    return (
      <Footer>
        <span className="upvote">
          {/* {postfooter.upVotes} */}
          {abbreviate(postfooter.upVotes, 1)} upvotes
        </span>
        {postfooter.Comments && (
          <span className="comment">
            {/* {postfooter.Comments} */}
            {abbreviate(postfooter.Comments, 1)} comments
          </span>
        )}
      </Footer>
    );
  }
};

export default PostFooter;
