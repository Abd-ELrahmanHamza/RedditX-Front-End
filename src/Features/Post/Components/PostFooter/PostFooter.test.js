import { shallow } from "enzyme";
import { screen, render, waitFor } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import darkTheme from "Theme/darkTheme";
import PostFooter from "./PostFooter";
import { BrowserRouter } from "react-router-dom";


describe("PostFooter Component", () => {
  const wrapper = shallow(<PostFooter  />);

  it("this is a test for PostFooter Component", () => {
    expect(wrapper).toMatchSnapshot();
  });

 
});
