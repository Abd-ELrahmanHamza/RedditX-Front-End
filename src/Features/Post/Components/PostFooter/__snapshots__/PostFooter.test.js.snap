// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooter Component this is a test for PostFooter Component 1`] = `
<styled.div>
  <styled.span
    onClick={[Function]}
  >
    <TfiComment />
     231 Comments
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <VscGift />
     Award
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <IoMdShareAlt />
     Share
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <TbBadge />
     Save
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <Styled(Dropdown)
      autoClose="outside"
      className="d-inline mx-2"
    >
      <DropdownToggle
        id="dropdown-autoclose-outside"
      >
        <HiOutlineDotsHorizontal />
      </DropdownToggle>
      <DropdownMenu
        flip={true}
      >
        <DropdownItem
          href="#"
        >
          <span>
            <BiHide />
             Hide
          </span>
        </DropdownItem>
        <DropdownItem
          href="#"
        >
          <span>
            <FiFlag />
             Report
          </span>
        </DropdownItem>
      </DropdownMenu>
    </Styled(Dropdown)>
  </styled.span>
</styled.div>
`;

exports[`VotingBar Component this is a test for PostFooter Component 1`] = `
<styled.div>
  <styled.span
    onClick={[Function]}
  >
    <TfiComment />
     231 Comments
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <VscGift />
     Award
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <IoMdShareAlt />
     Share
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <TbBadge />
     Save
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <Styled(Dropdown)
      autoClose="outside"
      className="d-inline mx-2"
    >
      <DropdownToggle
        id="dropdown-autoclose-outside"
      >
        <HiOutlineDotsHorizontal />
      </DropdownToggle>
      <DropdownMenu
        flip={true}
      >
        <DropdownItem
          href="#"
        >
          <span>
            <BiHide />
             Hide
          </span>
        </DropdownItem>
        <DropdownItem
          href="#"
        >
          <span>
            <FiFlag />
             Report
          </span>
        </DropdownItem>
      </DropdownMenu>
    </Styled(Dropdown)>
  </styled.span>
</styled.div>
`;

exports[`VotingBar Component this is a test for PostPublisher Component 1`] = `
<styled.div>
  <styled.span
    onClick={[Function]}
  >
    <TfiComment />
     231 Comments
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <VscGift />
     Award
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <IoMdShareAlt />
     Share
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <TbBadge />
     Save
  </styled.span>
  <styled.span
    onClick={[Function]}
  >
    <Styled(Dropdown)
      autoClose="outside"
      className="d-inline mx-2"
    >
      <DropdownToggle
        id="dropdown-autoclose-outside"
      >
        <HiOutlineDotsHorizontal />
      </DropdownToggle>
      <DropdownMenu
        flip={true}
      >
        <DropdownItem
          href="#"
        >
          <span>
            <BiHide />
             Hide
          </span>
        </DropdownItem>
        <DropdownItem
          href="#"
        >
          <span>
            <FiFlag />
             Report
          </span>
        </DropdownItem>
      </DropdownMenu>
    </Styled(Dropdown)>
  </styled.span>
</styled.div>
`;
